### PR TITLE
DDF-4178 updated cxf, camel, http, and shiro

### PR DESCRIPTION
**Pull request by stustison**
#### What does this PR do?
Update dependencies. Fixes a bug or two as well.
#### Who is reviewing it? 
@mcalcote @blen-desta @brjeter

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@coyotesqrl
@stustison

#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -301,12 +301,9 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-compiler-plugin</artifactId>
-                    <version>3.7.0</version>
+                    <version>3.8.0</version>
                     <configuration>
                         <compilerId>javac-with-errorprone</compilerId>
-                        <compilerArgs>
```
Comment (coyotesqrl) [inline pom.xml]: ❓ Are _all_ the junit issues that errorprone finds and reports now resolved?

Comment (stustison) [reply pom.xml]: The build runs correctly, but that flag no longer works on the new bundle plugin. I resolved all of the issues that that flag was hiding that caused the build to break.

File: catalog/core/catalog-core-directorymonitor/pom.xml [hunk 2]
```diff
@@ -255,6 +259,11 @@
             <groupId>commons-configuration</groupId>
             <artifactId>commons-configuration</artifactId>
             <version>${commons-configuration.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-configuration2</artifactId>
+            <version>${commons-configuration2.version}</version>
```
Comment (brjeter) [inline catalog/core/catalog-core-directorymonitor/pom.xml]: ❓ Does it still need the old version of `commons-configuration` above?

File: catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorIT.java [hunk 3]
```diff
@@ -311,6 +311,8 @@ private Option initContentDirectoryMonitorConfig() {
         new BundleInfo("commons-collections", "commons-collections"),
         new BundleInfo("org.apache.commons", "commons-collections4"),
         new BundleInfo("commons-configuration", "commons-configuration"),
+        new BundleInfo("org.apache.commons", "commons-configuration2"),
```
Comment (brjeter) [inline catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorIT.java]: ❓ Does it still need the old version of `commons-configuration` above?

**Comment by mcalcote** [comment]
We will need to release ddf-parent 1.0.1 (https://github.com/codice/ddf-parent/pull/1) and then update the parent dependency as part of this change right?

File: catalog/core/catalog-core-directorymonitor/pom.xml (hunk 2 above)
Comment (stustison) [reply catalog/core/catalog-core-directorymonitor/pom.xml]: Yes.

File: catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorIT.java (hunk 3 above)
Comment (stustison) [reply catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorIT.java]: Yes.

**Comment by stustison** [comment]
yeah we'll need to merge ddf-parent first, release it, and then merge this change

**Comment by stustison** [comment]
build now

**Comment by stustison** [comment]
build now

**Comment by stustison** [comment]
build now

File: catalog/pom.xml [hunk 4]
```diff
@@ -40,7 +40,7 @@
         -->
         <jaxb.annotate.plugin.version>0.6.0</jaxb.annotate.plugin.version>
         <jaxb.visitor.plugin.version>1.13</jaxb.visitor.plugin.version>
-        <catalog.joda.time.version>2.2</catalog.joda.time.version>
+        <catalog.joda.time.version>2.10</catalog.joda.time.version>
```
Comment (mcalcote) [inline catalog/pom.xml]: Would we want to use the `<joda-time.version>2.10</joda-time.version>` property from ddf-parent here?

File: features/pom.xml [hunk 5]
```diff
@@ -135,7 +134,7 @@
         <cxf.jboss.logging.version>3.3.0.Final</cxf.jboss.logging.version>
         <cxf.jdom.bundle.version>${jdom.bundle.version}</cxf.jdom.bundle.version>
         <cxf.jettison.version>1.4.0</cxf.jettison.version>
-        <cxf.joda.time.version>2.9.4</cxf.joda.time.version>
+        <cxf.joda.time.version>2.10</cxf.joda.time.version>
```
Comment (mcalcote) [inline features/pom.xml]: Noticed this was also the same version as above.

File: features/pom.xml [hunk 6]
```diff
@@ -124,7 +123,7 @@
         <cxf.hazelcast.version>3.9.2</cxf.hazelcast.version>
         <cxf.hibernate.validator.version>5.3.6.Final</cxf.hibernate.validator.version>
         <cxf.httpcomponents.asyncclient.version>4.1.3</cxf.httpcomponents.asyncclient.version>
-        <cxf.httpcomponents.core.version>4.4.9</cxf.httpcomponents.core.version>
+        <cxf.httpcomponents.core.version>4.4.10</cxf.httpcomponents.core.version>
```
Comment (mcalcote) [inline features/pom.xml]: Looks like we could also use https://github.com/codice/ddf-parent/blob/master/pom.xml#L52 here.

**Comment by stustison** [comment]
build now

**Pull request merged**